runtime/strategies/search-tokens-to-particles.js

```diff
@@ -30,11 +30,11 @@ module.exports = class SearchTokensToParticles extends Strategy {
   }
   async generate(strategizer) {
     let findParticles = token => this._byToken[token] || [];
-    var results = Recipe.over(strategizer.generated, new class extends RecipeWalker {
+    let generated = strategizer.generated.filter(result => !result.result.isResolved());
+    let terminal = strategizer.terminal;
+    var results = Recipe.over([...generated, ...terminal], new class extends RecipeWalker {
```

Comment:
wouldn't some of the "generated" be also "terminal", causing duplicate element in the array we iterate over on?

Reply:
I don't think so. We're operating on `this.generated` rather than `generated`, we'll find out if anything in `generated` is terminal after the next generation.

Reply:
I'm not sure i understand:

strategizer ctor:
  this._generated = []
  this._terminal = []

generation 0:
  run some strategies => generated = [plan1, plan2, plan3]

  then:
  this._generated = [plan1, plan2]
  this._terminal = []

generation 1:
   run some strategies => generated = [plan3 (parent=plan1)]

   then:
   this._generated = [plan1, plan2, plan3]
   this._terminal = [plan2]
   # so plan2 is in both - this._generated and this._terminal

generation 2:
  run search-tokens-to-particles strategy:
        let generated = strategizer.generated.filter(result => !result.result.isResolved());
	let terminal = strategizer.terminal;
	var results = Recipe.over([...generated, ...terminal] ….
  i think results will contain plan2 twice, won't it? or am i missing smth?

Reply:
I'm not sure if you're confusing `this._generated` with `this._population`.

I think in your example, after gen1 `_generated` would be `[plan3]` and `_terminal` would be `[plan2]`

Reply:
aaaaah, i see now, sorry about the confusion!